.github/workflows/deploy-staging.yaml

```diff
@@ -1,24 +1,22 @@
-name: Deploy to Staging Environment
+name: Deploy to Staging
 
 on:
+  # deploy when ci has been completed on main (should occur after new commits are added to main
+  # directly or via pull request)
   workflow_run:
     workflows: ["Continuous Integration (Python)"]
     types: [completed]
     branches:
       - main
-  workflow_dispatch:
```

Comment:
this will deploy everything when new pull requests are merged into main or if a rogue commit is added.

use the manual deploy script otherwise.